onemask/ArchitectureStudy/app/src/main/java/com/android/studyfork/ui/main/MainActivity.kt

```diff
@@ -0,0 +1,49 @@
+package com.android.studyfork.ui.main
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.android.studyfork.R
+import com.android.studyfork.ui.main.adapter.ViewPagerAdapter
+import com.android.studyfork.ui.main.presenter.MainContract
+import com.android.studyfork.ui.main.presenter.MainPresenter
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity(), MainContract.View {
+
+    private val viewPagerAdapter by lazy {
+        ViewPagerAdapter(
+            supportFragmentManager
+        )
+    }
+    private val presenter by lazy { MainPresenter(this) }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        initViewPager()
+        loadData()
+    }
+
+    private fun initViewPager() {
+        with(pager_content) {
+            adapter = viewPagerAdapter
```

Comment:
이런 식으로 해도 되지 않을까요? lazy를 사용하신 이유가 있나요?
```suggestion
            adapter = ViewPagerAdapter(supportFragmentManager)
```

Reply:
- viewPagerAdapter 변수를  lazy로 선언하면 setViewPagerData()에서 
```suggestion
 viewPagerAdapter.setData(titles,items)
```
  이렇게 선언할 수 있어서 깔끔해 보여서 lazy를 사용하게 되었습니다~